### PR TITLE
Qk8 s 903

### PR DESCRIPTION
**Pull request by subbuATB1**
Ingress changes for non-prod onPremise QK8s-903

### CODE REVIEW

**Comment by subbuATB1** [comment]
Lint does not fail in local however its getting failed in Git
C:\subbu\tricentis\code\helm\qtest-chart>helm lint Charts/qtest-chart
==> Linting Charts/qtest-chart

1 chart(s) linted, 0 chart(s) failed

**Comment by byangtri** [comment]
Validating maintainers...
41<https://github.com/QAS-Labs/qtest-chart/runs/4300204266?check_suite_focus=true#step:6:41>>>> git ls-remote --get-url origin
42<https://github.com/QAS-Labs/qtest-chart/runs/4300204266?check_suite_focus=true#step:6:42>>>> helm lint Charts/qtest-chart
43<https://github.com/QAS-Labs/qtest-chart/runs/4300204266?check_suite_focus=true#step:6:43>==> Linting Charts/qtest-chart
44<https://github.com/QAS-Labs/qtest-chart/runs/4300204266?check_suite_focus=true#step:6:44>Error:  templates/resourcequota.yaml: object name must be between 0 and 253 characters: ""

What you could do is possibly take the resourcequota from qtest-mgr helm chart as qtest-mgr is passing and qtest-chart is not.

Brandon Yang | DevOps Engineer


[http://signature.tricentis.com/Email_Signature_Tricentis_Generic_Footer.png]<https://www.tricentis.com/?utm_source=signature&utm_medium=email&utm_campaign=Tricentis%20Generic>

[http://signature.tricentis.com/Twitter.png]<https://twitter.com/Tricentis?ref_src=twsrc%5Egoogle%7Ctwcamp%5Eserp%7Ctwgr%5Eauthor>    [http://signature.tricentis.com/Linkedin.png] <https://www.linkedin.com/company/tricentis-technology-&-consulting-gmbh/>     [http://signature.tricentis.com/Blog.png] <https://www.tricentis.com/blog/>

Tricentis USA Corp.; 2570 W El Camino Real, Suite 540, Mountain View, CA 94040
This email (including any attachments) contains information that may be privileged and confidential. If you are not the intended recipient of this communication, please delete this e-mail and notify Tricentis immediately

________________________________

File: Charts/qtest-chart/templates/deployment.yaml [hunk 1]
```diff
@@ -90,19 +91,19 @@ spec:
...
             name: qtest-manager-config
+        - secretRef:
+            name: qtest-manager-secret
         env:
         - name: testconductor.environment
           value: {{ $uiProfile }}
         volumeMounts:
         - name: qtest-data
-          mountPath: /mnt/data/storage
+          mountPath: /mnt/host/c/mnt/data/storage
```
Comment (liemng) [inline Charts/qtest-chart/templates/deployment.yaml:101]: Can we have this in values file?  Linux won't have a C drive.

File: Charts/qtest-chart/templates/ingress.yaml [hunk 2]
```diff
@@ -1,61 +1,23 @@
...
-  annotations:
-    {{- toYaml . | nindent 4 }}
-  {{- end }}
-  {{- if .Values.ingress.labels }}
-  labels:
-  {{- range $key, $value := .Values.ingress.labels }}
-    {{ $key }}: {{ $value | quote }}
-  {{- end }}
-  {{- end }}
+  name: example-ingress
```
Comment (liemng) [inline Charts/qtest-chart/templates/ingress.yaml:4]: We should use ingress value name.

File: Charts/qtest-chart/templates/ingress.yaml [hunk 3]
```diff
@@ -1,61 +1,23 @@
...
+  namespace: qtest
```
Comment (liemng) [inline Charts/qtest-chart/templates/ingress.yaml:5]: We should use Release.Namespace.

File: Charts/qtest-chart/templates/ingress.yaml [hunk 4]
```diff
@@ -1,61 +1,23 @@
...
-                  number: {{ $servicePort }}
-  {{- end -}}
-  {{- end -}}
-  {{- if .Values.ingress.tls }}
-  tls:
-  {{ toYaml .Values.ingress.tls | nindent 4 }}
-    {{- end -}}
-  {{- end -}}
-
+  rules:
```
Comment (liemng) [inline Charts/qtest-chart/templates/ingress.yaml:7]: Don't use hard-coded values in template.

File: Charts/qtest-chart/templates/storageclass.yaml [hunk 5]
```diff
@@ -1,11 +1,11 @@
-{{ if and .Values.testconductor.environment.isOnPremise false }}
+{{ if and .Values.testconductor.environment.isOnPremise true }}
 kind: StorageClass
 apiVersion: storage.k8s.io/v1
 metadata:
   name: qtest-onpremise-storage
```
Comment (liemng) [inline Charts/qtest-chart/templates/storageclass.yaml:5]: This needs to be a value to allow user to specify the value.

File: Charts/qtest-chart/templates/storageclass.yaml [hunk 6]
```diff
@@ -14,13 +14,13 @@ metadata:
   namespace: {{ .Values.namespace.name }}
 spec:
   capacity:
-    storage: 5Gi
+    storage: 2Gi
```
Comment (liemng) [inline Charts/qtest-chart/templates/storageclass.yaml:17]: This needs to be in value.

File: Charts/qtest-chart/values-nonprod-op.yaml [hunk 7]
```diff
@@ -0,0 +1,205 @@
+### Ingress/IngressClass (> K8s 1.18-1.22+) #####
```
Comment (liemng) [inline Charts/qtest-chart/values-nonprod-op.yaml:1]: Please trim this file down to include only overrided values.

File: Charts/qtest-chart/values.yaml [hunk 8]
```diff
@@ -290,7 +291,7 @@ vpaAutoscaling:
 
 # HPA values
 autoscaling:
-  enabled: true
+  enabled: false
```
Comment (liemng) [inline Charts/qtest-chart/values.yaml:294]: Why is HPA disabled?

File: Charts/qtest-mgr/templates/storageclass.yaml [hunk 9]
```diff
@@ -1,3 +1,50 @@
+{{ if and .Values.testconductor.environment.isOnPremise true }}
```
Comment (liemng) [inline Charts/qtest-mgr/templates/storageclass.yaml:1]: Please don't have hard-coded values in template files.

File: Charts/qtest-chart/templates/deployment.yaml (hunk 1 above)
Comment (subbuATB1) [reply Charts/qtest-chart/templates/deployment.yaml:101]: Yes, I made the change to move volume mounts to values.yaml files.

File: Charts/qtest-chart/templates/ingress.yaml (hunk 2 above)
Comment (subbuATB1) [reply Charts/qtest-chart/templates/ingress.yaml:4]: This is in qtest-chart which will be deleted, I did use Ingress name from values file in qtest-mgr charts.

File: Charts/qtest-chart/templates/ingress.yaml (hunk 3 above)
Comment (subbuATB1) [reply Charts/qtest-chart/templates/ingress.yaml:5]: This is in qtest-chart which will be deleted, I did use Ingress name from values file in qtest-mgr charts.

File: Charts/qtest-chart/templates/ingress.yaml (hunk 4 above)
Comment (subbuATB1) [reply Charts/qtest-chart/templates/ingress.yaml:7]: This qtest-chart/Charts/qtest-charts will be decommissioned and we will be using qtest-charts/Charts/qtest-mgr will be used.

File: Charts/qtest-chart/templates/storageclass.yaml (hunk 5 above)
Comment (subbuATB1) [reply Charts/qtest-chart/templates/storageclass.yaml:5]: Storage class hardcode values moved to values.yaml file

File: Charts/qtest-chart/templates/storageclass.yaml (hunk 6 above)
Comment (subbuATB1) [reply Charts/qtest-chart/templates/storageclass.yaml:17]: Moved to values.yaml file

File: Charts/qtest-chart/values-nonprod-op.yaml (hunk 7 above)
Comment (subbuATB1) [reply Charts/qtest-chart/values-nonprod-op.yaml:1]: It does not work keep only overridden values , we have to keep all the values of a tree/object along with overridden values

File: Charts/qtest-mgr/templates/storageclass.yaml (hunk 9 above)
Comment (subbuATB1) [reply Charts/qtest-mgr/templates/storageclass.yaml:1]: storageclass hardcoded values moved to values.yaml file

File: Charts/qtest-chart/templates/deployment.yaml [hunk 10]
```diff
@@ -248,10 +247,10 @@ spec:
           value: {{ $pollerProfile }}
         volumeMounts:
         - name: qtest-data
-          mountPath: /mnt/data/storage
+          mountPath: /mnt/host/c/mnt/data/storage
```
Comment (isgo-golgo13) [inline Charts/qtest-chart/templates/deployment.yaml:250]: You need to get rid of the 'C' drive here. This is will explode on Linux machines. This value here needs to go in values-nonprod-op.yaml

File: Charts/qtest-chart/templates/deployment.yaml [hunk 11]
```diff
@@ -248,10 +247,10 @@ spec:
...
         {{- if $isOnPremise }}
         - name: qtest-onpremise-storage
-          mountPath: /mnt/data/qtest
+          mountPath: /mnt/host/c/mnt/data/qtest
```
Comment (isgo-golgo13) [inline Charts/qtest-chart/templates/deployment.yaml:253]: You need to get rid of the 'C' drive here. This is will explode on Linux machines. This value here needs to go in values-nonprod-op.yaml

File: Charts/qtest-chart/templates/ingress.yaml (hunk 3 above)
Comment (isgo-golgo13) [reply Charts/qtest-chart/templates/ingress.yaml:5]: Subby why are you doing the changes in qtest-chart ? I thought we were doing the changes in qtest-mgr?

File: Charts/qtest-chart/templates/storageclass.yaml [hunk 12]
```diff
@@ -14,13 +14,13 @@ metadata:
...
   accessModes:
     - ReadWriteMany
   persistentVolumeReclaimPolicy: Retain
   storageClassName: qtest-onpremise-storage
   local:
-    path: /mnt/data/qtest
+    path: /mnt/host/c/mnt/data/qtest
```
Comment (isgo-golgo13) [inline Charts/qtest-chart/templates/storageclass.yaml:23]: No C drive on Linux.  Your using WSL or MinGW locally?

**Review by isgo-golgo13 — CHANGES_REQUESTED** [review]
Revisit hardcoded 'C'drives in the paths

File: Charts/qtest-mgr/templates/deployment.yaml [hunk 13]
```diff
@@ -90,19 +91,19 @@ spec:
...
             name: qtest-manager-config
+        - secretRef:
+            name: qtest-manager-secret
         env:
         - name: testconductor.environment
           value: {{ $uiProfile }}
         volumeMounts:
         - name: qtest-data
-          mountPath: /mnt/data/storage
+          mountPath: /mnt/host/c/mnt/data/storage
```
Comment (liemng) [inline Charts/qtest-mgr/templates/deployment.yaml:101]: This needs to go into values.yaml, and should not have C drive in there.

File: Charts/qtest-mgr/templates/deployment.yaml [hunk 14]
```diff
@@ -90,19 +91,19 @@ spec:
...
         {{- if $isOnPremise }}
         - name: qtest-onpremise-storage
-          mountPath: /mnt/data/qtest
+          mountPath: /mnt/host/c/mnt/data/qtest
```
Comment (liemng) [inline Charts/qtest-mgr/templates/deployment.yaml:104]: Same comment as above

File: Charts/qtest-mgr/templates/deployment.yaml [hunk 15]
```diff
@@ -360,10 +359,10 @@ spec:
           value: {{ $notificationProfile }}
         volumeMounts:
         - name: qtest-data
-          mountPath: /mnt/data/storage
+          mountPath: /mnt/host/c/mnt/data/storage
```
Comment (liemng) [inline Charts/qtest-mgr/templates/deployment.yaml:362]: No values

File: Charts/qtest-mgr/templates/deployment.yaml [hunk 16]
```diff
@@ -360,10 +359,10 @@ spec:
...
         {{- if $isOnPremise }}
         - name: qtest-onpremise-storage
-          mountPath: /mnt/data/qtest
+          mountPath: /mnt/host/c/mnt/data/qtest
```
Comment (liemng) [inline Charts/qtest-mgr/templates/deployment.yaml:365]: No values

File: Charts/qtest-mgr/templates/deployment.yaml [hunk 17]
```diff
@@ -513,10 +514,10 @@ spec:
           value: {{ $apiProfile }}
         volumeMounts:
         - name: qtest-data
-          mountPath: /mnt/data/storage
+          mountPath: /mnt/host/c/mnt/data/storage
```
Comment (liemng) [inline Charts/qtest-mgr/templates/deployment.yaml:517]: No values

File: Charts/qtest-mgr/templates/deployment.yaml [hunk 18]
```diff
@@ -513,10 +514,10 @@ spec:
...
         {{- if $isOnPremise }}
         - name: qtest-onpremise-storage
-          mountPath: /mnt/data/qtest
+          mountPath: /mnt/host/c/mnt/data/qtest
```
Comment (liemng) [inline Charts/qtest-mgr/templates/deployment.yaml:520]: No values

File: Charts/qtest-mgr/templates/storageclass.yaml [hunk 19]
```diff
@@ -1,26 +1,73 @@
...
+kind: StorageClass
+apiVersion: storage.k8s.io/v1
+metadata:
+  name: {{ .Values.storage.name }}
+  namespace: {{ .Values.namespace.name }}
```
Comment (liemng) [inline Charts/qtest-mgr/templates/storageclass.yaml:6]: Don't need namespace.name.  Use built-in Helm objects:  https://helm.sh/docs/chart_template_guide/builtin_objects/ (Release.Namespace)

File: Charts/qtest-mgr/templates/storageclass.yaml [hunk 20]
```diff
@@ -1,26 +1,73 @@
...
+  local:
+    path: {{ .Values.volumeMounts.onpremise.mountPath }}
+  nodeAffinity:
+    required:
+      nodeSelectorTerms:
+        - matchExpressions:
+            - key: kubernetes.io/hostname
+              operator: In
+              values:
+                - kind-worker
```
Comment (liemng) [inline Charts/qtest-mgr/templates/storageclass.yaml:31]: Why do we have Kind stuff in the template?  Customer can run in AWS (and will).  Any Kind stuff should be moved into the values yaml file.

File: Charts/qtest-mgr/templates/storageclass.yaml [hunk 21]
```diff
@@ -1,26 +1,73 @@
...
+metadata:
+  name: {{ .Values.storage.pvc.name }}
+  namespace: {{ .Values.namespace.name }}
+spec:
+  accessModes:
+    - {{ .Values.storage.pv.accessmode }}
+  storageClassName: {{ .Values.storage.storageclass.name }}
+  resources:
+    requests:
+      storage: {{ .Values.storage.pv.size }}
```
Comment (liemng) [inline Charts/qtest-mgr/templates/storageclass.yaml:45]: Please use the correct value names as shown in the README file:  https://github.com/QAS-Labs/qtest-chart/blob/main/README.md

File: Charts/qtest-mgr/templates/storageclass.yaml [hunk 22]
```diff
@@ -1,26 +1,73 @@
...
+                - kind-worker2
+---
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+  name: {{ .Values.storage.pvc.name }}
+  namespace: {{ .Values.namespace.name }}
+spec:
+  accessModes:
+    - {{ .Values.storage.pv.accessmode }}
```
Comment (liemng) [inline Charts/qtest-mgr/templates/storageclass.yaml:41]: Please use the correct value names as shown in the README file: https://github.com/QAS-Labs/qtest-chart/blob/main/README.md

File: Charts/qtest-mgr/templates/storageclass.yaml [hunk 23]
```diff
@@ -34,13 +81,14 @@ spec:
...
 metadata:
-  name: qtest-onpremise-pvc
+  name: {{ .Values.storage.pvc.name }}
   namespace: {{ .Values.namespace.name }}
 spec:
   accessModes:
-    - ReadWriteMany
-  storageClassName: qtest-onpremise-storage
+    - {{ .Values.storage.pv.accessmode }}
+  storageClassName: {{ .Values.storage.storageclass.name }}
```
Comment (liemng) [inline Charts/qtest-mgr/templates/storageclass.yaml:89]: Add a condition in here.  If there is no storageClass defined, should not define it.  This is why your StorageClass does not work in Kind and you have to mount dirs by hand.

File: Charts/qtest-mgr/values-nonprod-op.yaml [hunk 24]
```diff
@@ -0,0 +1,219 @@
+### Ingress/IngressClass (> K8s 1.18-1.22+) #####
```
Comment (liemng) [inline Charts/qtest-mgr/values-nonprod-op.yaml:1]: Please include only values that you override.  Do not copy and paste everything.

**Review by liemng — CHANGES_REQUESTED** [review]
Please see my comments.  Also, please fix the lint issues and look at the value names in https://github.com/QAS-Labs/qtest-chart/blob/main/README.md and match them accordingly.

File: Charts/qtest-chart/values-nonprod-op.yaml (hunk 7 above)
Comment (liemng) [reply Charts/qtest-chart/values-nonprod-op.yaml:1]: That's not true.

File: Charts/qtest-chart/templates/deployment.yaml (hunk 1 above)
Comment (pjaladi21) [reply Charts/qtest-chart/templates/deployment.yaml:101]: Moved to values.yaml

File: Charts/qtest-chart/templates/storageclass.yaml (hunk 12 above)
Comment (pjaladi21) [reply Charts/qtest-chart/templates/storageclass.yaml:23]: Resolved

File: Charts/qtest-chart/templates/storageclass.yaml (hunk 5 above)
Comment (pjaladi21) [reply Charts/qtest-chart/templates/storageclass.yaml:5]: Moved to values.yaml

File: Charts/qtest-mgr/templates/deployment.yaml (hunk 13 above)
Comment (pjaladi21) [reply Charts/qtest-mgr/templates/deployment.yaml:101]: moved

File: Charts/qtest-mgr/templates/deployment.yaml (hunk 14 above)
Comment (pjaladi21) [reply Charts/qtest-mgr/templates/deployment.yaml:104]: Moved

File: Charts/qtest-mgr/templates/storageclass.yaml (hunk 19 above)
Comment (pjaladi21) [reply Charts/qtest-mgr/templates/storageclass.yaml:6]: This is already there so reusing

File: Charts/qtest-mgr/values-nonprod-op.yaml (hunk 24 above)
Comment (pjaladi21) [reply Charts/qtest-mgr/values-nonprod-op.yaml:1]: Can we leave this as is and get back to it later

Comment (pjaladi21) [reply Charts/qtest-mgr/values-nonprod-op.yaml:1]: Can we leave this as is and get back to it later

**Pull request closed without merging**